datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/build_traps_db.py

```diff
@@ -0,0 +1,294 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import json
+import os
+from contextlib import ExitStack
+from functools import lru_cache
+
+import click
+import yaml
+from pysmi import error
+from pysmi.codegen import JsonCodeGen
+from pysmi.compiler import MibCompiler
+from pysmi.parser import SmiV1CompatParser
+from pysmi.reader import getReadersFromUrls
+from pysmi.searcher import AnyFileSearcher
+from pysmi.writer import FileWriter
+
+from datadog_checks.dev import TempDir
+
+from ...console import CONTEXT_SETTINGS, abort, echo_failure, echo_info, echo_success, echo_warning
+from .constants import MIB_SOURCE_URL
+
+# Unique identifiers of traps in json-compiled MIB files.
+NOTIFICATION_TYPE = 'notificationtype'
+
+
+@click.command(
+    context_settings=CONTEXT_SETTINGS,
+    short_help='Generate a traps database that can be used by the '
+    'Datadog Agent for resolving Traps OIDs to readable strings.',
+)
+@click.option(
+    '--mib-sources',
+    '-s',
+    help='Url or a path to a directory containing the dependencies for [mib_files...].'
+    'Traps defined in these files are ignored.',
+)
+@click.option(
+    '--compiled-mibs-sources',
+    '-c',
+    help='Path to a directory where to store the intermediate compiled MIBs in JSON format. Defaults to a temp dir',
+)
+@click.option(
+    '--output-dir',
+    '-o',
+    help='Path to a directory where to store the created traps database file per MIB.'
+    'Recommended option, do not use with --output-file',
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
What is the use case for output dir vs output file?